safekeeper/src/wal_backup.rs

```diff
@@ -354,28 +349,35 @@ impl WalBackupTask {
 }
 
 pub async fn backup_lsn_range(
-    start_lsn: Lsn,
+    timeline: &Arc<Timeline>,
+    backup_lsn: &mut Lsn,
     end_lsn: Lsn,
     wal_seg_size: usize,
     timeline_dir: &Path,
     workspace_dir: &Path,
-) -> Result<Lsn> {
-    let mut res = start_lsn;
+) -> Result<()> {
+    let start_lsn = *backup_lsn;
     let segments = get_segments(start_lsn, end_lsn, wal_seg_size);
     for s in &segments {
         backup_single_segment(s, timeline_dir, workspace_dir)
             .await
             .with_context(|| format!("offloading segno {}", s.seg_no))?;
 
-        res = s.end_lsn;
+        let new_backup_lsn = s.end_lsn;
+        let res = timeline.set_wal_backup_lsn(new_backup_lsn);
+        if let Err(e) = res {
+            error!("failed to set wal_backup_lsn: {}", e);
+            return Err(e);
```

Comment:
It doesn't make sense to retry after canceled timeline, we can just terminate the whole task. OTOH, retrying is not  harmful, and AFAIR there is no adequate reaping of failed tasks, so can leave as is currently as well.

However, logging error in any case pointless as we do that above.

Reply:
> It doesn't make sense to retry after canceled timeline

Yes, but only if we rely on the fact that `set_wal_backup_lsn` can return error only when timeline is cancelled.

> However, logging error in any case pointless as we do that above.

Replaced with context()